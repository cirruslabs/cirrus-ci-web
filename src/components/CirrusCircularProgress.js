import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { cirrusColors } from '../cirrusTheme';
import { withStyles } from '@material-ui/core';

const styles = {
  progress: {
    color: cirrusColors.success,
  },
};

const CirrusCircularProgress = props => {
  return (
    <CircularProgress
      classes={{
        colorPrimary: props.classes.progress,
      }}
    />
  );
};

export default withStyles(styles)(CirrusCircularProgress);
