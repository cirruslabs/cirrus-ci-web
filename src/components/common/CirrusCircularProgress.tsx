import React from 'react';

import { useTheme } from '@mui/material';
import CircularProgress from '@mui/material/CircularProgress';

const CirrusCircularProgress = () => {
  let theme = useTheme();
  return <CircularProgress style={{ color: theme.palette.info.main }} />;
};

export default CirrusCircularProgress;
