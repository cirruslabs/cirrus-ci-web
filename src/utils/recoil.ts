// https://recoiljs.org/docs/guides/atom-effects/#local-storage-persistence
import { DefaultValue } from 'recoil';

export const localStorageEffect =
  key =>
  ({ setSelf, onSet }) => {
    const savedValue = localStorage.getItem(key);
    if (savedValue != null) {
      setSelf(JSON.parse(savedValue));
    }

    onSet(newValue => {
      if (newValue instanceof DefaultValue) {
        localStorage.removeItem(key);
      } else {
        localStorage.setItem(key, JSON.stringify(newValue));
      }
    });
  };
