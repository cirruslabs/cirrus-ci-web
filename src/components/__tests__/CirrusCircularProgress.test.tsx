import React from 'react';
import renderer from 'react-test-renderer';

import { RecoilRoot } from 'recoil';

import CirrusCircularProgress from 'components/common/CirrusCircularProgress';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <RecoilRoot>
        <CirrusCircularProgress />
      </RecoilRoot>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
