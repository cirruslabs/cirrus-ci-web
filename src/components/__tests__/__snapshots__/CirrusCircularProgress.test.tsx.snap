// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
  role="progressbar"
  style={
    Object {
      "color": "#0288d1",
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</span>
`;
